distributed/tests/test_core.py

```diff
@@ -640,13 +633,12 @@ async def connect_to_server():
 
     tasks = [asyncio.create_task(connect_to_server()) for _ in range(30)]
 
-    await asyncio.sleep(0)
-    assert pool._connecting
+    while not pool._connecting:
+        await asyncio.sleep(0.01)
+
     close_fut = asyncio.create_task(pool.close())
 
-    with pytest.raises(
-        CommClosedError, match="ConnectionPool not running. Status: Status.closed"
-    ):
+    with pytest.raises(asyncio.CancelledError):
```

Comment:
this could catch the gen_test CancelledError

Reply:
```python

    close_fut = asyncio.create_task(pool.close())

    done, _ = await asyncio.wait(tasks)
    await close_fut
    assert all(t.cancelled() for t in done)
```
has a similar effect, thanks